viewer/latexworkshop.ts

```diff
@@ -417,11 +417,11 @@ class LateXWorkshopPdfViewer implements ILatexWorkshopPdfViewer {
 
             // Since WebSockets are disconnected when PC resumes from sleep,
             // we have to reconnect. https://github.com/James-Yu/LaTeX-Workshop/pull/1812
-            const reconnect = (tries: number) => () => {
+            const reconnect = (tries: number) => async () => {
                 console.log(`Try to reconnect to LaTeX Workshop: (${tries}/10).`)
                 try {
                     this.connectionPort = createConnectionPort(this)
-                    this.connectionPort.onDidOpen(() => {
+                    await this.connectionPort.onDidOpen(() => {
```

Comment:
We have a `'await' has no effect on the type of this expression.ts(80007)` warning on this line. It is not advised to use `await` to force raising exceptions. You may need to use explicit try-catch inside.